File: 2026POTD/JANUARY/06-01-2026/1161.rb
def max_level_sum(root)
    q = [root]
    level = 1
    ans = 1
    max_sum = -Float::INFINITY

    while !q.empty?
        size = q.size
        sum = 0

        size.times do
            node = q.shift
            sum += node.val
            q << node.left if node.left
            q << node.right if node.right
        end

        if sum > max_sum
            max_sum = sum
            ans = level
        end
        level += 1
    end

    ans
end

File: 2026POTD/JANUARY/06-01-2026/1161.ts
function maxLevelSum(root: TreeNode | null): number {
    let q: TreeNode[] = [root!];
    let level = 1, ans = 1;
    let maxSum = -Infinity;

    while (q.length) {
        let size = q.length;
        let sum = 0;
        for (let i = 0; i < size; i++) {
            let node = q.shift()!;
            sum += node.val;
            if (node.left) q.push(node.left);
            if (node.right) q.push(node.right);
        }
        if (sum > maxSum) {
            maxSum = sum;
            ans = level;
        }
        level++;
    }
    return ans;
}